internal/providers/service.go

```diff
@@ -269,6 +270,31 @@ func (_ *providerService) ValidateGitHubInstallationId(ctx context.Context, toke
 	return nil
 }
 
+// GitHubAppInstallationDeletedPayload represents the payload of a GitHub App installation deleted event
+type GitHubAppInstallationDeletedPayload struct {
+	InstallationID int64 `json:"installation_id"`
+}
+
+func (p *providerService) DeleteGitHubAppInstallation(ctx context.Context, installationID int64) error {
+	installation, err := p.store.GetInstallationIDByAppID(ctx, strconv.FormatInt(installationID, 10))
```

Comment:
@eleftherias since the table seems to be GH app specific and GH app installation IDs are Int64, why was the installation ID column set as TEXT?

```sql
CREATE TABLE provider_github_app_installations (
    app_installation_id TEXT PRIMARY KEY,
```

Reply:
We created the tables before actually using them, and we missed this detail. I'm happy to create a migration to make it a BIGINT.

Reply:
let's do it in separate patch - I initially added one to the branch but as discussed elsewehere it would take too long to merge this PR. I'll send that as a separate PR and it lands when it lands.

Reply:
also added tests

Reply:
> also added tests

sorry wrong thread (unresolving so this one keeps being visible)

Reply:
I used `TEXT` despite it being GitHub-specific because I didn't want to attach too firmly to the implementation detail that these are int64 (for example, if we needed something different for GH Enterprise), but I'm happy to switch to int64 if we prefer.

I do think that there are a bunch of details to GitHub Apps that are probably unique, so using a column type that matches GitHub Apps closely doesn't really bother me here.